source/grid/manifold.cc

```diff
@@ -62,6 +62,128 @@ get_new_point (const Quadrature<spacedim> &quad) const
 }
 
 
+
+template <>
+Tensor<1,2>
+Manifold<2, 2>::
+normal_vector (const typename Triangulation<2, 2>::face_iterator &face,
+               const Point<2> &p) const
+{
+  const int spacedim=2, dim=2;
+
+  Tensor<1,spacedim> tangent = ((p-face->vertex(0)).norm_square() > (p-face->vertex(1)).norm_square() ?
+                                get_tangent_vector(p, face->vertex(0)) :
+                                -get_tangent_vector(p, face->vertex(1)));
+  return cross_product_2d(tangent);
```

Comment:
The documentation of `normal_vector` that was originally in `Boundary<dim,spacedim>` explicitly states that the vector need not be normalized, and that its direction w.r.t. inner or outer is unspecified.